### DISCUSSION TITLE
[react-query] Auto Refetch doesn't work when refetchInterval option is not set to over 0ms.

### DISCUSSION THREAD

**Discussion by sakupi01** [TanStack/query#6364] [Q&A] [answered]
After a mutation is executed and succeeded,  I invalidated a query that has a key related to the mutation as below.
```
const { mutate, error, isPending } = useMutation({
    mutationKey: [KEY],
    mutationFn: (body: BodyInit) =>
      fetcher({ url: KEY, postObj: { body: body, method: "POST" } }),
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ["/api/get/user"] });
    },
  });
```
In v4, after the invalidation of a specific query, regardless of how much time is set to the refetchInterval option of the query, invalidateQueries method lets us mark queries as stale and refetch them as well. 
Specifically, after the mutation to change the user name has been executed in a tab,  the user name is also updated on the other tab too. 
This behaviour is also demonstrated in the official mutation example [here](https://tanstack.com/query/v4/docs/react/examples/react/auto-refetching). (In the playground, if you open two tabs and either set the refetchInterval value 0 or remove the refetchInterval option out of useQuery, and then, you input a todo, the todo you input will be reflected in both tabs. )

However, in v5, despite I invalidated a specific query, it doesn't fetch the latest data or at least it's not reflected in UI.
This behaviour is also confirmed in the official mutation example. You can reproduce the issue by removing refetchInterval from useQuery and opening multiple tabs [here](https://tanstack.com/query/latest/docs/react/examples/react/auto-refetching). (In the playground, if you open two tabs and either set the refetchInterval value 0 or remove the refetchInterval option and input a todo, the todo you input will NOT be reflected in the other tab without reloading. )

It'd be awesome to be made clear the reason for this change, as this behaviour in v5 or the latest version leads to unnecessary network round trips (return 304 status code)!

**Comment by TkDodo** [comment, marked as answer]
> invalidateQueries method lets us mark queries as stale and refetch them as well.

that is still the case. note that only active queries are refetched immediately. that hasn't changed.

> Specifically, after the mutation to change the user name has been executed in a tab, the user name is also updated on the other tab too.


not sure what "other tab" means exactly. unless you have used the experimental `broadcast-channel` plugin, browser tabs don't know about each other.

note that there was a change to how [refetchOnWindowFocus](https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#window-focus-refetching-no-longer-listens-to-the-focus-event) works in v5. maybe you've just seen a refetch on window focus before?

**Reply by sakupi01** [reply]
@TkDodo 
didn't focus on  how [refetchOnWindowFocus](https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#window-focus-refetching-no-longer-listens-to-the-focus-event) works in v5, and the answer was perfect fit for me! 
broadcast-channel is also achieving what I wanted to do in the two different windows... 
Thanks!<3
